### PR TITLE
Fix CI for Fedora + some other things

### PR DESCRIPTION
**Pull request by iTrooz**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -376,12 +376,21 @@ jobs:
...
-  fedora-build-rawhide:
-    name: 🐧 Fedora Rawhide
+  fedora-build:
+    strategy:
+      matrix:
+        include:
+          - docker_image: fedora:latest
+            release: Latest
+          - docker_image: fedora:rawhide
+            release: RawHide
```
Comment (jonathanspw) [inline .github/workflows/build.yml:386]: "H" in "Rawhide" should be lowercase.  Seems silly but branding is important :)

**Comment by iTrooz** [comment]
Note : I didn't make too much effort with the commit names because they usually get squashed

**Comment by iTrooz** [comment]
Fixed the submodule problem

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -376,12 +376,21 @@ jobs:
...
-  fedora-build-rawhide:
-    name: 🐧 Fedora Rawhide
+  fedora-build:
+    strategy:
+      matrix:
+        include:
+          - docker_image: fedora:latest
+            release: Latest
+          - docker_image: fedora:rawhide
+            release: rawHide
```
Comment (jonathanspw) [inline .github/workflows/build.yml:386]: You made the R lowercase.  It should be "Rawhide".

**Review by jonathanspw — CHANGES_REQUESTED** [review]

**Comment by iTrooz** [comment]
I think this is ready to be merged !

This PR fix the Fedora CI build in various ways discussed on discord, and strips the binaries for all plateforms (I don't know how to test if this works for CPack tho)

**Comment by iTrooz** [comment]
I didn't know if MacOS had soversions so I only added them to Linux

**Comment by WerWolv** [comment]
Both Arch and Fedora don't seem to find libimhex anymore now. You probably forget to change the path somewhere

**Comment by iTrooz** [comment]
Okay, this time it should be good !

**Pull request merged**